### PR TITLE
feat: add plugin template config feature

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1512 
___

### CODE REVIEW

File: web/config/routes.ts [hunk 1]
```diff
@@ -109,6 +121,10 @@ const routes = [
     component: './User/Logout',
     layout: false,
   },
+  {
+    path: '/user/logout',
+    component: './User/Logout',
```
Comment (juzhiyuan) [inline web/config/routes.ts:126]: image

Duplicate Logout?

File: web/config/routes.ts [hunk 2]
```diff
@@ -99,6 +99,18 @@ const routes = [
     path: '/settings',
     component: './Setting',
   },
+  {
+    path: '/pluginTemplate/list',
```
Comment (juzhiyuan) [inline web/config/routes.ts:103]:
```suggestion
    path: '/plugin-template/list',
```

File: web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    cy.get('[title=Description]').type('test_plugin_template1');
+    cy.contains('button', 'Search').click();
+    cy.contains('test_plugin_template1').siblings().contains('Edit').click();
+
+    cy.get('#desc').clear().type('test_plugin_template2');
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+
+    cy.get(this.domSelector.notification).should('contain', 'Edit Plugin Template Successfully');
```
Comment (juzhiyuan) [inline web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js:66]: How to make sure data is updated successfully?

File: web/src/components/LabelsfDrawer/LabelsDrawer.tsx [hunk 4]
```diff
@@ -19,22 +19,23 @@ import { AutoComplete, Button, Col, Drawer, Form, notification, Row } from 'antd
...
   actionName: string;
   dataSource: string[];
+  filterList?: string[],
+  fetchLabelList: any,
   disabled: boolean;
   onClose: () => void;
 } & Pick<RouteModule.Step1PassProps, 'onChange'>;
 
-const LabelList = (disabled: boolean, labelList: RouteModule.LabelList) => {
+const LabelList = (disabled: boolean, labelList: LabelList, filterList: string[]) => {
```
Comment (juzhiyuan) [inline web/src/components/LabelsfDrawer/LabelsDrawer.tsx:34]:
```suggestion
const LabelList = (disabled: boolean, labelList: LabelList, filterList: string[] = []) => {
```

File: web/src/components/LabelsfDrawer/LabelsDrawer.tsx [hunk 5]
```diff
@@ -116,14 +117,16 @@ const LabelsDrawer: React.FC<Props> = ({
   actionName = '',
   disabled = false,
   dataSource = [],
+  filterList = [],
+  fetchLabelList,
```
Comment (juzhiyuan) [inline web/src/components/LabelsfDrawer/LabelsDrawer.tsx:121]: This variable is unused?

File: web/src/components/Plugin/PluginPage.tsx [hunk 6]
```diff
@@ -104,6 +115,54 @@ const PluginPage: React.FC<Props> = ({
...
+                  ].map((item) => (
+                    <Select.Option value={item.id!} key={item.id}>
+                      {item.desc}
+                    </Select.Option>
+                  ))}
+                </Select>
+              </Form.Item>
+            </Form>
+            <Alert message={<>
+              <p>{formatMessage({ id: 'component.plugin.pluginTemplate.tips1' })}</p>
```
Comment (juzhiyuan) [inline web/src/components/Plugin/PluginPage.tsx:161]: What's tips?

File: web/src/components/Plugin/locales/en-US.ts [hunk 7]
```diff
@@ -17,4 +17,8 @@
 export default {
   'component.plugin.tip1': 'NOTE: After customizing the plugin, you need to update schema.json.',
   'component.plugin.tip2': 'How to update?',
+  'component.select.pluginTemplate': 'Select PluginTemplate',
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/en-US.ts:20]:
```suggestion
  'component.select.pluginTemplate': 'Select a plugin template',
```

File: web/src/components/Plugin/locales/en-US.ts [hunk 8]
```diff
@@ -17,4 +17,8 @@
...
+  'component.step.select.pluginTemplate.select.option': 'Custom',
+  'component.plugin.pluginTemplate.tips1': '1.When a route already have plugins field configured, the plugins in the plugin template will be merged into it.',
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/en-US.ts:22]:
```suggestion
  'component.plugin.pluginTemplate.tip1': '1. When a route already have plugins field configured, the plugins in the plugin template will be merged into it.',
```

File: web/src/components/Plugin/locales/en-US.ts [hunk 9]
```diff
@@ -17,4 +17,8 @@
...
+  'component.plugin.pluginTemplate.tips2': '2.The same plugin in the plugin template will override one in the plugins'
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/en-US.ts:23]: ditto

File: web/src/components/Plugin/locales/zh-CN.ts [hunk 10]
```diff
@@ -17,4 +17,8 @@
 export default {
   'component.plugin.tip1': '注意：自定义插件后（修改、新增、删除等），需更新 schema.json。',
   'component.plugin.tip2': '如何更新？',
+  "component.select.pluginTemplate": '选择插件模板',
+  'component.step.select.pluginTemplate.select.option': '手动配置',
+  'component.plugin.pluginTemplate.tips1': '1.如果这个路由已经配置了插件，那么插件模板里面的插件配置会合并进去。',
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/zh-CN.ts:22]: ditto

Comment (juzhiyuan) [reply web/src/components/Plugin/locales/zh-CN.ts:22]: `若路由已配置插件，则插件模板数据将与已配置的插件数据合并。`

File: web/src/components/Plugin/locales/zh-CN.ts [hunk 11]
```diff
@@ -17,4 +17,8 @@
...
+  'component.plugin.pluginTemplate.tips2': '2.插件模板相同的插件会覆盖掉原有的插件。'
```
Comment (juzhiyuan) [inline web/src/components/Plugin/locales/zh-CN.ts:23]: ditto

File: web/src/pages/PluginTemplate/Create.tsx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+import { Card, Steps, notification, Form } from 'antd';
+import { history, useIntl } from 'umi';
+
+import ActionBar from '@/components/ActionBar';
+import PluginPage from '@/components/Plugin';
+
+import Step1 from './components/Step1';
+import Preview from './components/Preview';
+import { fetchItem, create, update, } from './service';
+import { transformLableValueToKeyValue } from '@/helpers';
```
Comment (juzhiyuan) [inline web/src/pages/PluginTemplate/Create.tsx:28]: This line should be ahead of Line25

**Review by juzhiyuan — APPROVED** [review]

**Comment by liuxiran** [comment]
LGTM expect that https://github.com/apache/apisix-dashboard/blob/77b42e839f8a79bbae8e0dd097195bf922076bbb/web/src/pages/Route/components/CreateStep4/CreateStep4.tsx#L63 also need to support select plugin template configuration

**Comment by LiteSun** [comment]
> LGTM expect that
> 
> https://github.com/apache/apisix-dashboard/blob/77b42e839f8a79bbae8e0dd097195bf922076bbb/web/src/pages/Route/components/CreateStep4/CreateStep4.tsx#L63
> 
> also need to support select plugin template configuration

good catch!

**Comment by LiteSun** [comment]
> https://github.com/apache/apisix-dashboard/blob/77b42e839f8a79bbae8e0dd097195bf922076bbb/web/src/pages/Route/components/CreateStep4/CreateStep4.tsx#L63
> 
> also need to support select plugin template configuration

updated

**Comment by liuxiran** [comment]
> > https://github.com/apache/apisix-dashboard/blob/77b42e839f8a79bbae8e0dd097195bf922076bbb/web/src/pages/Route/components/CreateStep4/CreateStep4.tsx#L63
> > 
> > also need to support select plugin template configuration
> 
> updated

LGTM👍, it seems that there are some thing wrong with Ci , maybe it need to force push to rerun.
and it would be better to add a test case to test for the content of step4 after pr merged thanks

**Review by juzhiyuan — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**